digdag-core/src/main/java/io/digdag/core/agent/RequireOperatorFactory.java

```diff
@@ -65,43 +68,149 @@ private RequireOperator(OperatorContext context, TaskCallbackApi callback)
+             *          - If ignore_failure is true or attempt finished successfully, require> op finished successfully
+             *          - else finished with exception.
+             */
             try {
-                StoredSessionAttempt attempt = callback.startSession(
+                projectIdentifier = Optional.of(makeProjectIdentifier());
+
+                callback.startSession(
                         context,
                         request.getSiteId(),
-                        projectId,
+                        projectIdentifier.get(),
                         workflowName,
                         instant,
                         retryAttemptName,
                         overrideParams);
-
-                boolean isDone = attempt.getStateFlags().isDone();
-                if (isDone) {
-                    if (!ignoreFailure && !attempt.getStateFlags().isSuccess()) {
-                        // ignore_failure is false and the attempt is in error state. Make this operator failed.
-                        throw new TaskExecutionException(String.format(ENGLISH,
-                                    "Dependent workflow failed. Session id: %d, attempt id: %d",
-                                    attempt.getSessionId(), attempt.getId()));
-                    }
-                    return TaskResult.empty(cf);
-                }
-                else {
-                    // TODO use exponential-backoff to calculate retry interval
-                    throw TaskExecutionException.ofNextPolling(1, ConfigElement.copyOf(request.getLastStateParams()));
-                }
+                throw nextPolling(request.getLastStateParams().deepCopy().set("require_kicked", true));
+            }
+            catch (SessionAttemptConflictException ex) {
+                return processAttempt(ex.getConflictedSession(), lastStateParams, rerunOn, ignoreFailure);
             }
             catch (ResourceNotFoundException ex) {
-                throw new ConfigException(ex);
+                throw new TaskExecutionException(String.format(ENGLISH, "Dependent workflow does not exist. %s, workflowName:%s",
+                        projectIdentifier.transform(ProjectIdentifier::toString).or(""), workflowName));
             }
             catch (ResourceLimitExceededException ex) {
                 throw new TaskExecutionException(ex);
             }
         }
+
+        private TaskResult processAttempt(StoredSessionAttempt attempt, Config lastStateParams, OptionRerunOn rerunOn, boolean ignoreFailure)
+        {
+            if (attempt.getStateFlags().isDone()) {
+                // A flag to distinguish whether the attempt is kicked by require> or previous attempt.
+                boolean requireKicked = lastStateParams.get("require_kicked", boolean.class, false);
+                if (!requireKicked &&  (
+                        rerunOn == OptionRerunOn.ALL ||
```

Comment:
When OptionRerunOn.ALL is set, the workflow should be executed regardless it was executed via `require` operator or not, right? If so, `rerunOn == OptionRerunOn.ALL` should not depend on `requireKicked`?

Reply:
Ah, `require_kicked` is marked on the current request not a previous attempt...

Reply:
Ah, got it. I had a wrong idea. It's for `Check attempt kicker` on the diagram.